sound/soc/sof/intel/hda.c

```diff
@@ -79,7 +79,8 @@ int hda_ctrl_dai_widget_setup(struct snd_soc_dapm_widget *w)
 	}
 
 	/* set HW_PARAMS flag */
-	config->flags = FIELD_PREP(SOF_DAI_CONFIG_FLAGS_MASK, SOF_DAI_CONFIG_FLAGS_HW_PARAMS);
+	config->flags = FIELD_PREP(SOF_DAI_CONFIG_FLAGS_CMD_MASK, SOF_DAI_CONFIG_FLAGS_HW_PARAMS) |
+			FIELD_PREP(SOF_DAI_CONFIG_FLAGS_QUIRK_MASK, quirk_flags);
```

Comment:
I don't think this works. FIELD_PREP will shift "quirk_flags" by 4 bits and e.g. 2_STEP_STOP_PAUSE is already defined as "BIT(4)", so this won't work.

Reply:
ack. thanks for the suggestion. fixed now